### PR TITLE
Public feed

### PR DESCRIPTION
**Pull request by Byambaa0325**
Added a servlet and datastore methods to display all messages stored in JSON format.

### CODE REVIEW

**Comment by Byambaa0325** [comment]
Added a page to display the messages

**Comment by huygaa11** [comment]
Hi! Thanks for sending the code in. I will do a little research by myself later today and then review the code

File: src/main/webapp/js/feed-loader.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+          const messageForm = document.getElementById('message-form');
+          messageForm.classList.remove('hidden');
+        }
+      });
+}
+
+// Fetch data and populate the UI of the page.
+function buildUI(){
+addLoginOrLogoutLinkToNavigation();
+fetchMessages();
```
Comment (huygaa11) [inline src/main/webapp/js/feed-loader.js:66]: fetchAndDisplayMessages()

File: src/main/webapp/js/feed-loader.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+showMessageFormIfLoggedIn();
```
Comment (huygaa11) [inline src/main/webapp/js/feed-loader.js:67]: I think displaying messages and submitting messages are 2 independent changes that should be in independent PRs (pull requests). Do you think you can create 2 PRs?

File: src/main/webapp/js/feed-loader.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
+// Fetch messages and add them to the page.
+function fetchMessages(){
+const url = '/feed';
```
Comment (huygaa11) [inline src/main/webapp/js/feed-loader.js:3]: Indenting looks a little off.

File: src/main/webapp/js/feed-loader.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  }
+  messages.forEach((message) => {
+   const messageDiv = buildMessageDiv(message);
+   messageContainer.appendChild(messageDiv);
+  });
+});
+}
+
+function buildMessageDiv(message){
+const usernameDiv = document.createElement('div');
```
Comment (huygaa11) [inline src/main/webapp/js/feed-loader.js:22]: here too, indenting looks off.

File: src/main/webapp/js/feed-loader.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+messageDiv.appendChild(bodyDiv);
+
+return messageDiv;
+}
+
+/**
+ * Shows the message form if the user is logged in and viewing their own page.
+ */
+function showMessageFormIfLoggedIn() {
+  fetch('/login-status')
```
Comment (huygaa11) [inline src/main/webapp/js/feed-loader.js:51]: Indenting is good here.

File: src/main/java/com/google/codeu/servlets/MessageFeedServlet.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.codeu.data.Datastore;
+import com.google.codeu.data.Message;
+import com.google.gson.Gson;
+
+/**
+* Handles fetching all messages for the public feed.
+*/
+
```
Comment (huygaa11) [inline src/main/java/com/google/codeu/servlets/MessageFeedServlet.java]: <nit> extra space

File: src/main/webapp/feed.html [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+<html>
+<head>
+<title>Message Feed</title>
+<link rel="stylesheet" href="/css/main.css">
+<link rel="stylesheet" href="/css/user-page.css">
+<script src = "/js/feed-loader.js"></script>
+<script src="/js/navigation-loader.js"></script>
+</head>
+<body onload="buildUI()">
+  <nav>
```
Comment (huygaa11) [inline src/main/webapp/feed.html:11]: In the future, we should refactor this out, so that new pages don't have to copy paste it anymore.

File: src/main/webapp/js/feed-loader.js [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+      .then((response) => {
+        return response.json();
+      })
+      .then((loginStatus) => {
+        if (loginStatus.isLoggedIn) {
+          const messageForm = document.getElementById('message-form');
+          messageForm.classList.remove('hidden');
```
Comment (huygaa11) [inline src/main/webapp/js/feed-loader.js:58]: Having "hidden" UI is bit risky in real life production. But it is okay for this project. :p

File: src/main/webapp/js/feed-loader.js (hunk 4 above)
Comment (huygaa11) [reply src/main/webapp/js/feed-loader.js:22]: basically, indent everything in a method so that it is easy to see where the method ends.

**Review by huygaa11 — CHANGES_REQUESTED** [review]

File: src/main/webapp/js/feed-loader.js (hunk 8 above)
Comment (Byambaa0325) [reply src/main/webapp/js/feed-loader.js:58]: Noted

File: src/main/webapp/feed.html (hunk 7 above)
Comment (Byambaa0325) [reply src/main/webapp/feed.html:11]: Noted

File: src/main/webapp/js/feed-loader.js (hunk 1 above)
Comment (Byambaa0325) [reply src/main/webapp/js/feed-loader.js:66]: it is the project default function name

**Comment by Byambaa0325** [comment]
Closing to do a stacked pull request
